exporters/secret/exporter.go

```diff
@@ -102,14 +111,69 @@ func (e *Exporter) calculateExpiry(ch chan<- prometheus.Metric, secret v1.Secret
 
 		for _, cert := range certs {
 			timestamp := float64(cert.NotAfter.Unix())
-			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, certType)
+			issuer := cert.Issuer.String()
+			ch <- prometheus.MustNewConstMetric(e.cert, prometheus.GaugeValue, timestamp, secretName, secretNamespace, certKey, issuer, certCRName)
 		}
 	}
 	e.logger.Log("info", fmt.Sprintf("added secret %s/%s to the metrics", secretNamespace, secretName))
 
 	return nil
 }
 
+func (e *Exporter) certificateGroupVersionResources() ([]schema.GroupVersionResource, error) {
+	groupVersions := []schema.GroupVersionResource{}
+
+	_, resources, err := e.k8sClient.ServerGroupsAndResources()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	for _, rl := range resources {
+		for _, r := range rl.APIResources {
+			if r.Name == "certificates" && strings.HasPrefix(rl.GroupVersion, "cert-manager.io/") {
+				s := schema.FromAPIVersionAndKind(rl.GroupVersion, r.Name)
+				groupVersions = append(groupVersions, schema.GroupVersionResource{
+					Group:    s.Group,
+					Version:  s.Version,
+					Resource: r.Name,
+				})
+			}
+		}
+	}
+
+	return groupVersions, nil
+}
+
+func (e *Exporter) findCertCRName(secretName, secretNamespace string) (string, error) {
+	certificateGroupVersionResources, err := e.certificateGroupVersionResources()
+	if err != nil {
+		e.logger.Log("error", microerror.Mask(err))
+		return "", err
+	}
+	if len(certificateGroupVersionResources) == 0 {
+		e.logger.Log("info", "cert-manager Certificate not available, skipping secret collection")
```

Comment:
```suggestion
		e.logger.Log("info", "cert-manager Certificate custom resource definition not available, skipping secret collection")
```

I think this is more accurate while debugging. You do not even have the CRD if I understood the code correctly.